static/src/javascripts/projects/common/modules/article/space-filler.js

```diff
@@ -1,66 +1,47 @@
+     * A safer way of using spacefinder.
+     * Given a set of spacefinder rules, applies a writer to the first matching paragraph.
+     * Uses fastdom to avoid layout-thrashing, but queues up asynchronous writes to avoid race conditions. We don't
+     * seek a slot for a new component until all the other component writes have finished.
+     *
+     * @param rules - a spacefinder ruleset
+     * @param writer - function, takes a para element and injects a container for the new content synchronously. It should NOT use Fastdom.
+     *
+     * @returns {Promise} - when insertion attempt completed, resolves 'true' if inserted, or 'false' if no space found
+     */
+    function fillSpace(rules, writer) {
+        return lastIteration = lastIteration.then(insertNextContent).catch(onInsertionError);
 
-        /**
-         * A safer way of using spacefinder.
-         * Given a set of spacefinder rules, applies a writer to the first matching paragraph.
-         * Uses fastdom to avoid layout-thrashing, but queues up asynchronous writes to avoid race conditions. We don't
-         * seek a slot for a new component until all the other component writes have finished.
-         *
-         * @param rules - a spacefinder ruleset
-         * @param writer - function, takes a para element and injects a container for the new content synchronously. It should NOT use Fastdom.
-         * @param debug - flag to enable debugging in spacefinder
-         *
-         * @returns {Promise} - when insertion attempt completed, resolves 'true' if inserted, or 'false' if no space found
-         */
-        this.insertAtFirstSpace = function insertAtFirstSpace(rules, writer, debug) {
-            lastInsertion = lastInsertion.then(insertNextContent).catch(onInsertionError);
+        function insertNextContent() {
+            return spacefinder.findSpace(rules).then(insertionPromise);
+        }
 
-            function insertNextContent() {
-                return spacefinder.getParaWithSpace(rules, debug).then(function applyParaToWriter(para) {
-                    if (para) {
-                        return insertionPromise(para);
-                    } else {
-                        return false;
-                    }
-                });
-            }
-
-            function insertionPromise(para) {
-                return new Promise(function (resolve, reject) {
-                    idleFastdom.write(function () {
-                        try {
-                            writer(para);
-                            resolve(true);
-                        } catch (e) {
-                            reject(e);
-                        }
-                    });
-                });
-            }
-
-            function onInsertionError(e) {
-                raven.captureException(e);
-                return false;
-            }
+        function insertionPromise(slots) {
+            return fastdom.write(function () {
+                return writer(slots);
```

Comment:
I'm not sure if I would reject the promise in that case; I think the idiom is more for reject to be used for exceptions, and this isn't something I'd normally use `throw` for. But I guess it is fine to choose our own promise interface so long as we document it.